### ISSUE TITLE
New kind of display for the albums list

### ISSUE THREAD

**Issue by lazka** [quodlibet/quodlibet#241]
Original [issue 241](https://code.google.com/p/quodlibet/issues/detail?id=241) created by brunobergot on 2009-06-24T17:40:51.000Z:

Hi and sorry for my poor english,

I really like the look and feel of the albums list. This list can be very
long when you have a lot of albums, and generally i like to search an album
only by scanning the covers &quot;au hasard&quot;.

I think it would be great to allow users to modifiy some settings about the
display of this list :
- show textuals informations or not
- show this information when the mouse hover a cover
- size of the covers

Some years ago i was using a plugin called album list for winamp. I would
love to have this type of display in the album list of QL.

Thanks for reading

**Comment by brunob** [comment]
Work is on in my fork here : https://bitbucket.org/b_b_/quodlibet

**Comment by brunob** [comment]
Switched my work here : https://github.com/brunob/quodlibet/tree/covergrid

**Comment by brunob** [comment]
See also #1489 (in order to centralize issues about this feature).

**Comment by Lalopx** [comment]
Is there a way to change the album view, or add the option to the right clivk menu?? i would love to help with some coding. Also, hello to evryone :+1:

**Comment by darthoctopus** [comment]
@brunob's fork seems to have died. I've taken the liberty of getting covergrid to work with 3.7.1, but only after brutally commenting out a bunch of calls to GtkIconView.set_cursor. Would you be open to a pull request for covergrid as-is?

**Comment by brunob** [comment]
@darthoctopus i haven't touched to my fork since a while cause i never knowed if it would be merge one day... I think it would be better to ask to @lazka to have a look at this, maybe a pull request of your fork would suit to him.

**Comment by nodiscc** [comment]
Duplicate of https://github.com/quodlibet/quodlibet/issues/1101?
Related https://github.com/quodlibet/quodlibet/issues/1665.

**Comment by brunob** [comment]
@nodiscc #1101 is the duplicate of thie one present here ;)

**Comment by nodiscc** [comment]
I've just drafted what an album art based view should look at in my humble opinion. Feedback welcome.

![ql-albugrid](https://cloud.githubusercontent.com/assets/3950390/19727202/c257fca4-9b8e-11e6-9c5b-961686d67135.png)
![ql-hover](https://cloud.githubusercontent.com/assets/3950390/19727203/c29725dc-9b8e-11e6-9231-c4e6c2af5303.png)

Clicking anywhere on an album art selects it and display details in the left pane.
Selecting multiple albums displays 'N albums selected in the left pane, along with play all/enqueue all buttons.

**Comment by nodiscc** [comment]
About https://github.com/quodlibet/quodlibet/issues/1665, a variant of this view could be used

 * Display a top-level artists list/images grid instead of directly an album grid
 * Upon clicking on the artist, the album grid for this artist is displayed, along with a `Back` button to clear the selected artist.
 * Selecting an album works as described above.

**Comment by darthoctopus** [comment]
well @brunob already did the heavy lifting to implement this (with less fancy), I've been using it on a regular basis for a while now and thought it would be a shame not to merge this to master, is all. Right now we show the songlist in an RVPaned display (i.e. horizontal instead of vertical split) but I guess that can be left to some config variable (not sure how to implement though).

**Comment by brunob** [comment]
@darthoctopus i also like to see this one merged, since i asked for it in 2099 and passed a lot of time on last year (this was clearly not easy for me since it was the first time i made ptyhon + gtk code ^^). 

If i remember correctly, @lazka had some doubts about the performances of this display, but QL core surely have evolved since my last commits, and all this may need a big update.

**Comment by urielz** [comment]
just writing to express that as a QL user this is one of the missing features that I would like to see implemented as well. so +1 to get it merged.

**Comment by declension** [comment]
Can we close this ticket too now?

**Comment by brunob** [comment]
I think so.
